activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -243,7 +243,7 @@ def owner_key_type
           end
 
           def reflection_scope
-            @reflection_scope ||= reflection.join_scopes(klass.arel_table, klass.predicate_builder, klass).inject(&:merge!) || klass.unscoped
+            @reflection_scope ||= reflection.join_scopes(klass.arel_table, klass.predicate_builder).inject(klass.unscoped, &:merge!)
```

Comment:
My bad, there was a sublle difference I missed the `klass` being passed as third param.